numcodecs/pcodec.py

```diff
@@ -44,39 +47,70 @@ class PCodec(Codec):
     def __init__(
         self,
         level: int = 8,
+        mode_spec: Literal["auto", "classic"] = "auto",
+        delta_spec: Literal["auto", "none", "try_consecutive", "try_lookback"] | None = None,
```

Comment:
I realize the difference between "none" and None could be confusing - how about we make the default "auto"? And then "auto" + delta_encoding_order is the legacy case to handle.